examples/api/lib/material/toggle_buttons/toggle_buttons.0.dart

```diff
@@ -0,0 +1,171 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flutter code sample for ToggleButtons
+
+import 'package:flutter/material.dart';
+
+const List<Widget> fruits = <Widget>[
+  Text('Apple'),
+  Text('Banana'),
+  Text('Orange')
+];
+
+const List<Widget> vegetables = <Widget>[
+  Text('Tomatoes'),
+  Text('Potatoes'),
+  Text('Carrots')
+];
+
+const List<Widget> icons = <Widget>[
+  Icon(Icons.sunny),
+  Icon(Icons.cloud),
+  Icon(Icons.ac_unit),
+];
+
+void main() => runApp(const MyApp());
+
+class MyApp extends StatelessWidget {
+  const MyApp({Key? key}) : super(key: key);
+
+  static const String _title = 'ToggleButtons Sample';
+
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      title: _title,
+      home: ToggleButtonsSample(title: _title),
+    );
+  }
+}
+
+class ToggleButtonsSample extends StatefulWidget {
+  const ToggleButtonsSample({Key? key, required this.title}) : super(key: key);
+
+  final String title;
+
+  @override
+  State<ToggleButtonsSample> createState() => _ToggleButtonsSampleState();
+}
+
+class _ToggleButtonsSampleState extends State<ToggleButtonsSample> {
+  final List<bool> _selectedFruits = <bool>[true, false, false];
+  final List<bool> _selectedVegetables = <bool>[false, true, false];
+  final List<bool> _selectedWeather = <bool>[false, false, true];
```

Comment:
Same here: since this is a single-selection case, it too can be an index and simplify the code below.